_extensions/quarto-ext/shinylive/shinylive.lua

```diff
@@ -90,8 +93,15 @@ return {
           quarto.doc.attach_to_dependency("shinylive", dep)
         end
 
-        el.attr.classes = pandoc.List()
-        el.attr.classes:insert("shinylive-python")
+        if el.attr.classes:includes("{shinylive-python}") then
+          el.attributes.engine = "python"
+          el.attr.classes = pandoc.List()
+          el.attr.classes:insert("shinylive-python")
+        elseif el.attr.classes:includes("{shinylive-r}") then
+            el.attributes.engine = "r"
+            el.attr.classes = pandoc.List()
+            el.attr.classes:insert("shinylive-r")
```

Comment:
Is the different indentation here intentional?

Reply:
Whoops! No, not sure what happened there.

I've amended the commit so this block's indentation matches the block above.